pkg/engine/armvariables.go

```diff
@@ -340,9 +335,9 @@ func getK8sMasterVars(cs *api.ContainerService) (map[string]interface{}, error)
 		} else {
 			masterVars["primaryAvailabilitySetName"] = ""
 		}
-		masterVars["primaryScaleSetName"] = ""
 		masterVars["vmType"] = "standard"
 	}
+	masterVars["primaryScaleSetName"] = cs.Properties.GetPrimaryScaleSetName()
```

Comment:
We can just set this variable using the getter now that the getter returns an empty string if the primary agent pool is not vmss.